protocol/activity_center_persistence.go

```diff
@@ -60,10 +60,6 @@ func (db sqlitePersistence) DeleteActivityCenterNotificationForMessage(chatID st
 	}
 
 	for _, notification := range notifications {
-		if notification.LastMessage != nil && notification.LastMessage.ID == messageID {
-			withNotification(notification)
-		}
-
```

Comment:
This logic was not included in the [initial implementation](https://github.com/status-im/status-go/pull/2319) done by @briansztamfater  but was later introduced in [this PR](https://github.com/status-im/status-go/pull/2854) by @rasom. I'm uncertain whether removing it will cause any issues that I'm not aware of. To the best of my knowledge, everything seems to be working correctly based on my testing. Please don't hesitate to inform me if I've overlooked any details.

Reply:
This check compares the `LastMessage.ID` of the notification with the ID of the deleted message. When deleting the last sent message, the ID of the deleted message matches the `LastMessage.ID` of all other messages in the activity center, causing the removal of all notifications. This is the reason I removed this part, as it resolves the issue.

Reply:
I don't fully understand the implications, but doesn't deleting this code means we'll have a notification pointing to a orphaned message that has been marked to be deleted? See https://github.com/status-im/status-go/blob/eb437e9d8dd5bd1ca3b18fa05c4853d6ae8f4956/protocol/messenger_handler.go#L1870

Reply:
@ajayesivan, GitHub sent me a notification to review this PR again. I can't be sure if the fix is good or not from the code alone, and since you tested, seems alright.